src/TestFramework/Coverage/JUnit/JUnitTestCaseSorter.php

```diff
@@ -59,11 +59,11 @@ public function getUniqueSortedFileNames(array $tests): iterable
             /** @var TestLocation $testLocation */
             $testLocation = current($uniqueTestLocations);
 
+            /** @var string $filePath */
             $filePath = $testLocation->getFilePath();
 
-            if ($filePath !== null) {
-                yield $filePath;
-            }
+            // Path for TestFileTimeData is not optional. It is never a null.
```

Comment:
I would slightly tweak the comment to reflect that then otherwise the comment is confusing as it contradicts the type

Reply:
Also might be more appropriate to put L62 with the type

Reply:
Can you see if it is better now?